File: main_project/client/java/OceAdapter/src/main/java/com/xiaonei/xce/feed/FeedEdmCacheAdapter.java
package com.xiaonei.xce.feed;

import java.util.Map;

import xce.edm.FeedEdmCachePrx;
import xce.edm.FeedEdmCachePrxHelper;
import xce.util.ng.channel.ChannelFactory;
import xce.util.ng.channel.ChannelFactory.ChannelType;
import xce.util.ng.proxyholder.AbstractModProxyHolder;
import Ice.ObjectPrx;

public class FeedEdmCacheAdapter extends
		AbstractModProxyHolder<Integer, FeedEdmCachePrx> {
	private static FeedEdmCacheAdapter instance;
	private static String endpoints = "@FeedEdmCache";

	public FeedEdmCacheAdapter() {
		super(ChannelFactory.getChannel(ChannelType.XCEFEED), "FEC"
				+ endpoints, 1);
	}

	public static FeedEdmCacheAdapter getInstance() {
		if (instance == null) {
			instance = new FeedEdmCacheAdapter();
		}
		return instance;
	}

	@Override
	protected int mod(Integer key, int size) {
		return key % size;
	}

	@Override
	public FeedEdmCachePrx uncheckedCast(ObjectPrx baseProxy) {
		return FeedEdmCachePrxHelper.uncheckedCast(baseProxy);
	}

	@Override
	protected ObjectPrx initProxy(ObjectPrx baseProxy) {
		return baseProxy;
	}

	private FeedEdmCachePrx getManagerOneway(int uid) {
		return uncheckedCast(getProxy(uid).ice_oneway());
	}

	private FeedEdmCachePrx getManager(int uid) {
		return getProxy(uid);
	}
  
  public void SetSendingFlag(int edm_id) throws Exception {
    getManagerOneway(edm_id).SetSendingFlag(edm_id);
  } 
  public void SetSuspendFlag(int edm_id) throws Exception {
    getManagerOneway(edm_id).SetSuspendFlag(edm_id);
  }
  public void UpdateEdmInfo(int edm_id)  throws Exception {
    getManagerOneway(edm_id).UpdateEdmInfo(edm_id);
  }
  public void SetForComplete()  throws Exception {
    getManager(0).SetForComplete();
  }
}

File: main_project/OceCxxAdapter/src/FeedEdmCacheAdapter.cpp
#include "FeedEdmCacheAdapter.h"
namespace xce {
namespace edm {

FeedEdmCachePrx FeedEdmCacheAdapter::getManager(int id) {
  return locate<FeedEdmCachePrx> (_managers, "FEC", id, MyUtil::TWO_WAY);
}

FeedEdmCachePrx FeedEdmCacheAdapter::getManagerOneway(int id) {
  return locate<FeedEdmCachePrx> (_managersOneway, "FEC", id, MyUtil::ONE_WAY);
}
FeedSeedInfoPtr FeedEdmCacheAdapter::SelectEdm(Ice::Int user_id, const MyUtil::Str2StrMap& params, const FeedUserInfoPtr& user) {
  return getManager(user_id)->SelectEdm(user_id,params,user);
}
FeedSeedInfoPtr FeedEdmCacheAdapter::SelectEdmForPhone(Ice::Int user_id, const MyUtil::Str2StrMap& params, const FeedUserInfoPtr& user) {
  return getManager(user_id)->SelectEdmForPhone(user_id,params,user);
}
FeedSeedInfoPtr FeedEdmCacheAdapter::SelectEdmById(int edmid) {
  return getManager(0)->SelectEdmById(edmid);
}
FeedSeedInfoPtr FeedEdmCacheAdapter::SelectEdmByType(int userid, int type) {
  return getManager(0)->SelectEdmByType(userid, type);
}
std::string FeedEdmCacheAdapter::LookUpEdm(Ice::Int edm_id) {
  return getManager(edm_id)->LookUpEdm(edm_id);
}
void FeedEdmCacheAdapter::SetSendingFlag(Ice::Int edm_id) {
  getManagerOneway(edm_id)->SetSendingFlag(edm_id);
}
void FeedEdmCacheAdapter::SetSuspendFlag(Ice::Int edm_id) {
  getManagerOneway(edm_id)->SetSuspendFlag(edm_id);
}
void FeedEdmCacheAdapter::UpdateEdmInfo(Ice::Int edm_id) {
  getManagerOneway(edm_id)->UpdateEdmInfo(edm_id);
}
void FeedEdmCacheAdapter::SetForComplete() {
  getManagerOneway(0)->SetForComplete();
}
}
}

